aptos-move/e2e-move-tests/src/tests/stake.rs

```diff
@@ -188,6 +198,75 @@ fn test_staking_rewards() {
         get_stake_pool(&harness, &validator_1_address).active,
         stake_amount_1
     );
+
+    // Enable rewards rate decrease and change rewards config. In production it requires governance.
+    let core_resources =
+        harness.new_account_at(AccountAddress::from_hex_literal("0xA550C18").unwrap());
+    let script_code = PROPOSAL_SCRIPTS
+        .get("enable_rewards_rate_decrease")
+        .expect("proposal script should be built");
+    let txn = harness.create_script(&core_resources, script_code.clone(), vec![], vec![]);
+    assert_success!(harness.run(txn));
+
+    // Parameters from the proposal.
+    let one_year_in_micros: u64 = 365 * 24 * 60 * 60 * 1_000_000;
+    let one_year_in_secs: u64 = one_year_in_micros / 1_000_000;
+    let mut rewards_rate: u64 = 100;
+    let min_rewards_rate: u64 = 30;
+    let rewards_rate_denominator: u64 = 10000;
+    let rewards_rate_decrease_rate_bps: u64 = 5000;
+    let bps_denominator: u64 = 10000;
+
+    // Both validators propose a block in the current epoch. Both should receive rewards at the
+    // new rewards rate, 1% every epoch.
+    harness.new_block_with_metadata(validator_1_address, vec![]);
+    harness.new_block_with_metadata(validator_2_address, vec![]);
+    harness.new_epoch();
+    stake_amount_1 += stake_amount_1 * rewards_rate / rewards_rate_denominator;
+    stake_amount_2 += stake_amount_2 * rewards_rate / rewards_rate_denominator;
+    assert_eq!(
+        get_stake_pool(&harness, &validator_1_address).active,
+        stake_amount_1
+    );
+    assert_eq!(
+        get_stake_pool(&harness, &validator_2_address).active,
+        stake_amount_2
+    );
+
```

Comment:
Can you add a 6 month check here to ensure the reward rate doesn't change before the 1 year mark is hit?

Reply:
Added!